pghoard/webserver.py

```diff
@@ -83,36 +196,35 @@ def __init__(self, config, requested_basebackup_sites, compression_queue, transf
         self._running = False
         self.log.debug("WebServer initialized with address: %r port: %r", self.address, self.port)
         self.is_initialized = threading.Event()
+        self.prefetch_404 = deque(maxlen=32)  # pylint: disable=attribute-defined-outside-init
+        self.download_results_processor = DownloadResultsProcessor(
+            self.lock, self.log, self.download_results, self.pending_download_ops, self.prefetch_404
+        )
```

Comment:
This introduced a new Thread inside another thread (WebServer), but it would not be shut down gracefully when PgHoard quits. It should be managed by the PgHoard main thread instead of WebServer, which is similar to how the main Thread manages the WebServer thread.  See `pghoard/pghoard.py::quit` method.

Reply:
I've added teardown of this new tread at `WebServer.close`.
https://github.com/Aiven-Open/pghoard/blob/3484a046ec44a408874052fabe88442d6ffcf8d6/pghoard/webserver.py#L227

I can move this call to pghoard:quit method, but if I launch thread in Webserver it better stops it in the same place (class).

Reply:
That is not what I meant. When PgHoard quits, it should wait for all threads to stop and quit first (join). That is what `pghoard/pghoard.py::quit` does. 

But in the above new code, when PgHoard quits, the `DownloadResultsProcessor` just sets a stop flag (setting running=False does not mean the thread is shutdown, it could be still in the middle of something), and the main thread does not wait for `DownloadResultsProcessor` is stopped.

Reply:
It's a good point!
It should be enough to add `self.download_results_processor.join()` to `WebServer.close`
I don't want to add this new thread to `PGHoard._get_all_threads` because it added to object unnecessary attributes.
https://github.com/Aiven-Open/pghoard/blob/f64ffb51aa79fa155de66662d6f4039da5fb4aaf/pghoard/pghoard.py#L971
What do you think?

Reply:
I think it should work from a function perspective, but it breaks the pattern of how PgHoard handles all sub-threads. 

On the other hand, the download processing thread can be independent of the WebServer thread and does not have to be under WebServer, meanwhile, it might make troubleshooting/statistics more difficult.